SRVKS-1243: Add Serving transport encryption docs (#82435)
openshift/openshift-docs: modules/serving-transport-encryption-configuring.adoc

> :_content-type: PROCEDURE

Reviewer: missing metadata

[id="serving-transport-encryption-configuring_{context}"]
= Configuring transport encryption

Configuring transport encryption consists of two parts:

. Specifying ClusterIssuer to use:

`clusterLocalIssuerRef`: issuer for cluster-local-domain certificates used for ingress.

`systemInternalIssuerRef`: issuer for certificates for system-internal-tls certificates used by Knative internal components.

> . Specifying transport encryption features to use:

Reviewer 2: Not sure if this will be different in the final docs but the rendered start counting the bullets from 1 again. I believe this bullet should have number `2.`.  Screenshot:
![image](https://github.com/user-attachments/assets/6dce74a1-7a31-439e-85e5-407875c99207)

Author: Fixed the markup.

. Apply the `KnativeServing` resource:
+
[source,terminal]
----
$ oc apply -f <filename>
----

. Optionally, change the `defaultCertificate` in the Ingress Controller:
+
[source,yaml]
----
apiVersion: operator.openshift.io/v1
kind: IngressController
 ...
spec:
  defaultCertificate:
    name: ca-ingress-cert
----

. If you changed the `defaultCertificate`, you need to specify the custom certificate name in the `openshift-ingress-default-certificate` field in the `KnativeServing` custom resource. For example, if the custom certificate name is `ca-ingress-cert`, add the following configuration:
+
[source,yaml]
----
...
spec:
  config:
    network:
      system-internal-tls: Enabled
      openshift-ingress-default-certificate: "ca-ingress-cert"
...
----

. Restart the Controller component if you enabled `cluster-local-domain-tls` or `system-internal-tls`:
+
[IMPORTANT]
====
When either the `cluster-local-domain-tls` or the `system-internal-tls` feature is enabled, the Controller component needs to be restarted to enable the Knative Serving `cert-manager` integration.
====
+
[source,terminal]
----
$ oc rollout restart deploy/controller -n knative-serving
----

> . Restart the Activator component if you enabled `system-internal-tls`:
> +
> [IMPORTANT]
> ====
> When the `system-internal-tls` feature is activated, the Activator component needs to be restarted to reconfigure its internal web server, as this is not possible during runtime.
> ====

Reviewer 3: same here
```suggestion
. If you enabled `system-internal-tls`, restart the Activator component by running the following command.
+
[IMPORTANT]
====
When the `system-internal-tls` feature is activated, you must restart the Activator component to reconfigure its internal web server, as this is not possible during runtime.
====
```

`cluster-local-domain-tls`: Enables the transport encryption feature for cluster-local domains

`system-internal-tls`: Enables the transport encryption feature for OpenShift Serverless Serving internal components.

.Procedure

. Enable `transport-encryption` in `KnativeServing`:

[source,yaml]
----
apiVersion: operator.knative.dev/v1beta1
kind: KnativeServing
metadata:
  name: knative-serving
  namespace: knative-serving
spec:
  ...
  config:
    certmanager:
      clusterLocalIssuerRef: |
        kind: ClusterIssuer
        name: knative-serving-ca-issuer <1>
      systemInternalIssuerRef: |
        kind: ClusterIssuer
        name: knative-serving-ca-issuer <2>
    network:
      cluster-local-domain-tls: Enabled <3>
      system-internal-tls: Enabled <4>
----
<1> Define the ClusterIssuer for each feature. The same or individual ClusterIssuers can be used.
<2> Define the ClusterIssuer.
<3> Enabling the `cluster-local-domain-tls` feature. This and other features can be enabled or disabled individually.
<4> Enabling the `system-internal-tls` feature.

. Apply the `KnativeServing` resource:

[source,terminal]
----
$ oc apply -f <filename>
----

> 

Reviewer 2: Ideally, we'd describe how to change the default Ingress certificate here as described in https://github.com/openshift/openshift-docs/pull/79108#discussion_r1786116937

Author: Done in https://github.com/openshift/openshift-docs/pull/82435/commits/361a5c167989bae5689c59c9dd3881d7007352aa

Reviewer 2: This part looks good to me!

Author: @mgencur Thanks! What about the section as a whole?
